errors/errors.go

```diff
@@ -0,0 +1,130 @@
+package errors
+
+// Error represents all errors which can be passed between services
+type Error interface {
+	Code() string
+	Error() string
+	Type() ErrorType
+}
+
+// ErrorType is an enumerated type of error
+type ErrorType string
+
+const (
+	// ErrUnknown indicates an unknown type of error
+	// @todo should this just be mapped to an internal server error?
+	ErrUnknown = ErrorType("UnknownError")
```

Comment:
I agree with the TODO. As far as Im concerned everything unexpected is an InternalServerError